markdown/spring-cloud-stream-beyond/spring-cloud-stream-beyond.md

```diff
@@ -672,6 +678,143 @@ You'll note that you received 1 message on the default binding destination of `m
 
 ![Batch Publish 2](img/batchPublishTryMe2.webp)
 
+## Batch Consumption
+Duration: 0:07:00
+
+Batch consumers are a feature that allows consuming messages in batches rather than individually. This feature is particularly useful when dealing with high-volume data processing scenarios where processing messages individually may introduce performance overhead.
+
+Batching of messages is applied only when the `spring.cloud.stream.bindings.<binding-name>.consumer.batch-mode` is set to true. The number of messages in a batch is dependent on the availability of messages in the queue and the timeout to receive messages from the queue.
+
+Other parameters that can control the batching are:
+
+`batch-max-size`
+The maximum number of messages per batch.
+Only applicable when `batch-mode` is `true`, and the default value is 255
+
+
+`batch-timeout`
+The maximum wait time in milliseconds to receive a batch of messages. If this timeout is reached, then the messages that have already been received will be used to create the batch. A value of `0` means wait forever, and the default value is 5000
+Only applicable when `batch-mode` is `true`.
+
+
+It should be noted that a batch that the binder creates is a collection of individual messages and must not to be treated as a single consistent unit. 
+
+### Batch Consumer in action
+👀 Let's check it out! 
+``` java 
```

Comment:
Please tab all code to the left one to prevent extra space in code snippet

Reply:
Fixed.